imod/mf6/utilities/regrid.py

```diff
@@ -107,11 +91,31 @@ def get_regridder(
         """
         regridder_class = self.__get_regridder_class(regridder_type)
 
-        if not self.__has_regridder(regridder_class, method):
-            self.__create_regridder(regridder_class, method)
+        if "layer" not in source_grid.coords and "layer" in target_grid.coords:
+            target_grid = target_grid.drop_vars("layer")
+            
+        source_hash  = get_grid_geometry_hash(source_grid)
+        target_hash  = get_grid_geometry_hash(target_grid) 
+        key = (source_hash, target_hash, regridder_class)       
```

Comment:
The regridder cache is accesible to users. When regridding a model or a simulation, the user doesn;t need to interact with it, but when regridding a package, the cache is a function argument. So users can create one in their own scripts and use it as they please. The target hash can be used to prevent misuse